backend/src/mastodon/timeline.ts

```diff
@@ -10,10 +9,44 @@ export async function pregenerateTimelines(domain: string, db: D1Database, cache
 	await cache.put(actor.id + '/timeline/home', timeline)
 }
 
+type ActorFollowingResult = {
+	id: string
+	actorFollowersURL: string | null
+}
+
 export async function getHomeTimeline(domain: string, db: D1Database, actor: Actor): Promise<Array<MastodonStatus>> {
-	const following = await getFollowingId(db, actor)
+	const { results: following } = await db
+		.prepare(
+			`
+            SELECT
+                actor_following.target_actor_id as id,
+                json_extract(actors.properties, '$.followers') as actorFollowersURL
+            FROM actor_following
+            INNER JOIN actors ON actors.id = actor_following.target_actor_id
+            WHERE actor_id=? AND state='accepted'
+        `
+		)
+		.bind(actor.id.toString())
+		.all<ActorFollowingResult>()
+
+	let followingIds: string[] = []
+	let followingFollowersURLs: string[] = []
+
+	if (following) {
+		followingIds = following.map((row: ActorFollowingResult) => row.id)
+		followingFollowersURLs = following.map((row: ActorFollowingResult) => {
+			if (row.actorFollowersURL) {
+				return row.actorFollowersURL
+			} else {
+				// We don't have the Actor's followers URL stored, we'll guess
+				// one.
+				return row.id + '/followers'
+			}
+		})
```

Comment:
```suggestion
		followingFollowersURLs = following.map(({actorFollowersURL}: ActorFollowingResult) => 
			actorFollowersURL ?? `${row.id}/followers`
		)
```
also as I mentioned above the type may not be needed?

Reply:
Removed